virtool/users/db.py

```diff
@@ -78,3 +81,100 @@ async def compose_groups_update(
         raise ResourceConflictError(f"Non-existent groups: {', '.join(repr_ids)}")
 
     return {"groups": group_ids}
+
+
+async def update_mongo_user(
```

Comment:
@ReeceHoffmann recommended it to reduce the complexity of the update function, given we eventually will be no longer on mongo.

Reply:
We will leave this as is, seeing as it will soon be deleted.

This pattern for building updates comes up frequently. You are going to have to write dual support for MongoDB + Postgres in the future. Follow this pattern:

* Check `if field in update request`
  * Add to Mongo `update`.
  * Update Postgres model object.
* Check `if field in update request`
  * Add to Mongo `update`.
  * Update Postgres model object.
* Check `if field in update request`
  * Add to Mongo `update`.
  * Update Postgres model object.
* Commit changes
  * Call `update_one` for Mongo.
  * Nothing required for Postgres if you are in `both_transactions`. Commit will happen when context manager closes.

This lets us see the parallel actions happening across the two databases. Everything happening is right in front of our faces.

---

Here is the current form:

* Check `if field in update request`
  * Update Postgres model object.
* Check `if field in update request`
  * Update Postgres model object.
* Check `if field in update request`
  * Update Postgres model object.
* Commit changes
  * Call another function that copies exactly the above series of conditionals but does the Mongo-flavored actions. There is also a giant interface for this function that not only takes the Mongo and Postgres objects, but the entire users data object.
  * Nothing required for Postgres if you are in `both_transactions`. Commit will happen when context manager closes.

The one argument for this implementation is you can just delete the Mongo function and its call when Mongo support is dropped. I don't think that justifies the indirection and addition of another function.

I don't want to see the users data object being passed around by dependency injection. Right now we have largely maintained a nice layered structure:

API -> Data -> direct database access

Functions that do direct database access only need references to the database client objects (`mongo`, `pg`). In `update_mongo_user`, we are not only taking the database clients, but the entire user data object. This muddles our hierarchy and pushes us back into spaghetti territory.

Hope this makes sense.